### ISSUE TITLE
Plugin works only once?

### ISSUE THREAD

**Issue by arvinddoraiswamy** [jpederson/Squirrel.js#25]
I finally got it working (Issue #24) but now think I have a new problem - well 2 actually.

a) This works for the first time I have it on a form (Add feedback) and saves values beautifully. Then I do other stuff on my site and click Add feedback again. Then I start typing again - this time nothing is saved, for some reason. Something else I need to do?

b) While probably **not** an issue, what all "going away" actions are considered here? As in.. if I go type half the form, go away and come back and click on my "Add new feedback" link again - will squirrel have saved half the form and should it re-populate? Coz it doesn't.

Lastly, I feel guilty opening up Issues each time - when I mostly just need  bit of help getting it working, is this the best way to get in touch though?

**Comment by jpederson** [comment]
Hello! Yes, this is a fine way to get help, because if someone else has the
same issue, they can likely find their solution here.

That said, I'm not in the office at the moment, but will look into this
further when I get home.

**Comment by softwarespot** [comment]
What does your form look like? Does your inputs have name attributes or id attributes associated with each input? Do you have a submit button? Are there any errors showing in the console? What browser are you using?

Could you perhaps create an example on JSFiddle.net so we can look at it to determine if it's the plugin or not?

Edit: In the meantime, I have updated the example by adding a submit button. This was just to test that I didn't break anything whilst updating the plugin.

**Comment by arvinddoraiswamy** [comment]
Here is a jsfiddle URL - https://jsfiddle.net/kvwz6o95/

Console doesn't show any errors (well 1 but that's for a different JS and due to some other dependency). Nothing on this form.

It works well the first time. Then the record is saved. Then I come back and try and create another record. This time nothing is saved :(

Tested on Firefox and Chrome - both on Ubuntu 14.04.

**Comment by softwarespot** [comment]
I am not seeing what you're seeing, [see](https://gist.github.com/softwarespot/f69282adfcbe51c84f07) You can verify it's working by looking at the resources tab and the session tree item. If you're not running on a local web server, the sometimes you will have to click out and back to see the values change. But I added text, refresh, it was still there. I submitted the form and repeated the process. I simply can't replicate on my setup. Are you using the latest repo version?

**Comment by arvinddoraiswamy** [comment]
I am yeah. Do I also need to load the css file? Coz right now all I loaded
was the squirrel.js.

**Comment by softwarespot** [comment]
No, check out the Gist in my previous post of a barebones example.

**Comment by arvinddoraiswamy** [comment]
Oh well.. must be something I am screwing up then. I can't imagine what.
The JS btw is loaded at the start (in my app) with all the others. Is that
a problem? As in, do I have to add <script src> in every file I call
squirrel.js?

**Comment by jpederson** [comment]
Normally, it's best practice for any non-integral JS to be loaded at the bottom of the page, because the browser will block while it's loading script tags (unless they're loaded asynchronously, which I doubt is the case here). For example, important things to load right away would be an HTML5 Shim, Respond.js, Modernizr, or anything else that the CSS uses to render properly. I always load jQuery and Squirrel at the very end of the file. This is for performance reasons, because then the content will all be there before any scripts load and try to execute.

That said, it's _only_ a performance concern - Squirrel waits until the page is loaded to execute in order to ensure that any forms on the page are complete. That's done with the `jQuery.ready()` method, as can be found at the bottom of the main script file - so it wouldn't hurt anything if it was loaded in the head using the Rails asset pipeline like you probably are.

**Comment by arvinddoraiswamy** [comment]
I think I now have this working on all my forms except one. That's almost certainly not due to the plug-in but more of my ignorance :). I think it's clear that the plugin is working (and it is an awesome plugin thanks again).

If I continue to have trouble with this specific form, I will post a new issue.

**Comment by softwarespot** [comment]
Okay. Well if you can come up with a reproducer please let us know. Maybe we haven't accounted for every possible situation that can happen with a form.

**Comment by arvinddoraiswamy** [comment]
No its fine, I figured it out. Its just all the way Rails calls its JS files that were the problem. The plugin is great and works everywhere now. Thanks for the help.

**Comment by softwarespot** [comment]
No problem.
